services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/internal/document/DefaultEnhancedDocument.java

```diff
@@ -92,13 +92,21 @@ public static <T> AttributeConverter<T> converterForClass(EnhancedType<T> type,
                                                               ChainConverterProvider chainConverterProvider) {
 
         if (type.rawClass().isAssignableFrom(List.class)) {
+            List<EnhancedType<?>> params = type.rawClassParameters();
+            if (params.isEmpty()) {
+                throw new IllegalStateException("Type parameter is missing for " + type);
+            }
             return (AttributeConverter<T>) ListAttributeConverter
-                .create(converterForClass(type.rawClassParameters().get(0), chainConverterProvider));
+                .create(converterForClass(params.get(0), chainConverterProvider));
         }
         if (type.rawClass().isAssignableFrom(Map.class)) {
+            List<EnhancedType<?>> params = type.rawClassParameters();
+            if (params.isEmpty()) {
+                throw new IllegalStateException("Type parameters are missing for " + type);
+            }
             return (AttributeConverter<T>) MapAttributeConverter.mapConverter(
-                StringConverterProvider.defaultProvider().converterFor(type.rawClassParameters().get(0)),
-                converterForClass(type.rawClassParameters().get(1), chainConverterProvider));
+                StringConverterProvider.defaultProvider().converterFor(params.get(0)),
+                converterForClass(params.get(1), chainConverterProvider));
```

Comment:
Is it possible to end up in a state where the rawClassParameters only contains one element? Should we check that it has 2 entries?

Reply:
I think client code can do that, so I will add check for size